pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/auth/FunctionAuthProvider.java

```diff
@@ -39,29 +40,22 @@
 
     /**
      * Cache auth data in as part of function metadata for function that runtime may need to configure authentication
-     * @param tenant tenant that the function is running under
-     * @param namespace namespace that is the function is running under
-     * @param name name of the function
+     * @param funcDetails the function details
      * @param authenticationDataSource auth data
      * @return
      * @throws Exception
      */
-    Optional<FunctionAuthData> cacheAuthData(String tenant, String namespace, String name, AuthenticationDataSource authenticationDataSource) throws Exception;
-
-
-
-    Optional<FunctionAuthData> updateAuthData(String tenant, String namespace, String name, Optional<FunctionAuthData> existingFunctionAuthData, AuthenticationDataSource authenticationDataSource) throws Exception;
+    Optional<FunctionAuthData> cacheAuthData(Function.FunctionDetails funcDetails, AuthenticationDataSource authenticationDataSource) throws Exception;
```

Comment:
The changes to these methods allow us the flexibility we need to have implementations of this interface, primarily the `KubernetesSecretTokenAuthProvider`, have all context it needs to make sure that secrets it generates match the kubenetes namespace the function actually gets deployed too. 

This also seems like a reasonable thing to do so that implementations of this interface can have more context about the function for custom implementations